plotsearch/serializers/plot_search.py

```diff
@@ -1013,14 +1016,69 @@ class DirectReservationLinkSerializer(serializers.ModelSerializer):
     targets = InstanceDictPrimaryKeyRelatedField(
         queryset=PlotSearchTarget.objects.all(), many=True
     )
+    lang_choices = ("FI", "SE", "EN")
+    language = serializers.ChoiceField(lang_choices, allow_blank=True, required=False)
+    first_name = serializers.CharField(allow_blank=True, required=False)
+    last_name = serializers.CharField(allow_blank=True, required=False)
+    email = serializers.CharField(allow_blank=True, required=False)
+    company = serializers.CharField(allow_blank=True, required=False)
+    covering_note = serializers.CharField(allow_blank=True, required=False)
+    send_copy = serializers.BooleanField(required=False)
+    send_mail = serializers.BooleanField(required=False)
 
     class Meta:
         model = DirectReservationLink
         fields = (
             "uuid",
             "url",
             "targets",
+            "language",
+            "first_name",
+            "last_name",
+            "email",
+            "company",
+            "covering_note",
+            "send_copy",
+            "send_mail",
         )
 
     def get_url(self, obj):
         return obj.get_external_url()
+
+    def create(self, validated_data):
+        language = validated_data.pop("language")
+        first_name = validated_data.pop("first_name")
+        last_name = validated_data.pop("last_name")
+        email = validated_data.pop("email")
+        company = validated_data.pop("company")
+        covering_note = validated_data.pop("covering_note")
+        send_copy = validated_data.pop("send_copy")
+        send_email = validated_data.pop("send_mail")
+        instance = super().create(validated_data)
+
+        if send_email:
+            receivers = [email]
```

Comment:
Should there be comma?

Reply:
If I remember correctly, black was ranting against it. In tuples comma is needed to make a difference between regular parenthesis